pkg/ebpf/c/tracee.bpf.c

```diff
@@ -2830,6 +2833,81 @@ int tracepoint__sched__sched_process_exit(struct bpf_raw_tracepoint_args *ctx)
+{
+    event_data_t data = {};
+    if (!init_event_data(&data, ctx))
+        return 0;
+
+    struct socket** new_sock = bpf_map_lookup_elem(&accept_socket_new, &data.context.host_tid);
+
+    struct socket** old_sock = bpf_map_lookup_elem(&accept_socket_old, &data.context.host_tid);
+    bpf_map_delete_elem(&accept_socket_new, &data.context.host_tid);
+    bpf_map_delete_elem(&accept_socket_old, &data.context.host_tid);
+
+    if (new_sock == NULL) {
+        return -1;
+    }
+    if (old_sock == NULL) {
+        return -1;
+    }
+
+    struct sock *sk_new = get_socket_sock(*new_sock);
+    struct sock *sk_old = get_socket_sock(*old_sock);
+
+    u16 family_old = get_sock_family(sk_old);
+    u16 family_new = get_sock_family(sk_new);
+
+    if (((family_old != AF_UNIX) && (family_old != AF_INET) && (family_old != AF_INET6)) ||
+        ((family_new != AF_UNIX) && (family_new != AF_INET) && (family_new != AF_INET6))) {
+        return 0;
+    }
+
+    if (family_old == AF_INET && family_new == AF_INET) {
+        net_conn_v4_t net_details_old = {};
+        struct sockaddr_in local;
+        get_network_details_from_sock_v4(sk_old, &net_details_old, 0);
+        get_local_sockaddr_in_from_network_details(&local, &net_details_old, family_old);
+
+        save_to_submit_buf(&data, (void *)&local, sizeof(struct sockaddr_in), 1);
+
+        net_conn_v4_t net_details_new = {};
+        struct sockaddr_in remote;
+        get_network_details_from_sock_v4(sk_new, &net_details_new, 0);
+        get_remote_sockaddr_in_from_network_details(&remote, &net_details_new, family_new);
+
+        save_to_submit_buf(&data, (void *)&remote, sizeof(struct sockaddr_in), 2);
+    }
+    else if (family_old == AF_INET6 && family_new == AF_INET6) {
+        net_conn_v6_t net_details_old = {};
+        struct sockaddr_in6 local;
+        get_network_details_from_sock_v6(sk_old, &net_details_old, 0);
+        get_local_sockaddr_in6_from_network_details(&local, &net_details_old, family_old);
+
+        save_to_submit_buf(&data, (void *)&local, sizeof(struct sockaddr_in6), 1);
+
+        net_conn_v6_t net_details_new = {};
+
+        struct sockaddr_in6 remote;
+        get_network_details_from_sock_v6(sk_new, &net_details_new, 0);
+        get_remote_sockaddr_in6_from_network_details(&remote, &net_details_new, family_new);
+
+        save_to_submit_buf(&data, (void *)&remote, sizeof(struct sockaddr_in6), 2);
+    }
+    else if (family_old == AF_UNIX && family_new == AF_UNIX) {
+
+        struct unix_sock *unix_sk_new = (struct unix_sock *)sk_new;
+        struct sockaddr_un sockaddr_new = get_unix_sock_addr(unix_sk_new);
+        save_to_submit_buf(&data, (void *)&sockaddr_new, sizeof(struct sockaddr_un), 1);
+
+        struct unix_sock *unix_sk_old = (struct unix_sock *)sk_old;
+        struct sockaddr_un sockaddr_old = get_unix_sock_addr(unix_sk_old);
+        save_to_submit_buf(&data, (void *)&sockaddr_old, sizeof(struct sockaddr_un), 2);
+    }
```

Comment:
i think you should add `else` and return. otherwise you submit the event without saved args on the buffer.
what do you think?